src/Components/Scorecard.js

```diff
@@ -48,11 +57,95 @@ const Scorecard = () => {
     })
   }
 
+  const nextHole = (prevHoleScores) => {
+    const hole = scorecard.layout.holes.find(hole => hole.hole_number === currentHole.number + 1)
+    if(hole.hole_number === scorecard.layout.holes.length){
+      setInProgress(false)
+    }
+    setCurrentHole({
+      number: hole.hole_number,
+      players: updatePlayerScores(prevHoleScores),
+      par: hole.par,
+      distance: hole.distance
+    })
+  }
+
+  const updatePlayerScores = (prevHoleScores) => {
+    const updatedPlayers = {}
+    prevHoleScores.forEach(player => {
+      updatedPlayers[player.player_id] = { 
+        name: currentHole.players[player.player_id].name,
+        id: currentHole.players[player.player_id].id,
+        score: 0,
+        totalScore: player.score + currentHole.players[player.player_id].totalScore
+      }
+    })
+    return updatedPlayers
+  }
+
+  const postHoleScores = () => {
+    const currentPlayers = Object.keys(currentHole.players)
+    const playerScores = currentPlayers.map(player_id => {
+      return {
+        player_id: currentHole.players[player_id].id, 
+        strokes: currentHole.players[player_id].score
+      }
+    })
+    
+    const body = {
+      hole: {
+        hole_number: currentHole.number,
+        player_scores: playerScores
+      }
+    }
+    scoreAHole(body, scorecard.roundId)
+    .then(data => nextHole(data.scores))
+    .catch(error => setErrorMessage(error))
+  }
+
+  const postFinalScores = () => {
+    const currentPlayers = Object.keys(currentHole.players)
+    const playerScores = currentPlayers.map(player_id => {
+      return {
+        player_id: currentHole.players[player_id].id, 
+        strokes: currentHole.players[player_id].score
+      }
+    })
+    
+    const body = {
+      hole: {
+        hole_number: currentHole.number,
+        player_scores: playerScores
+      }
+    }
+    scoreAHole(body, scorecard.roundId)
+    .then(() =>  endCurrentRound())
+    .catch(error => setErrorMessage(error))
+  }
+
+  const endCurrentRound = () => {
+    endRound(scorecard.roundId)
+    .then(data => {
+      setScorecard({
+        ...scorecard,
+        final: data
```

Comment:
We could probable make some other context to set instead of stuffing this into scorecard